tests/test_openclhpp.cpp

```diff
@@ -516,8 +512,9 @@ void testMoveAssignContextNonNull(void);
 void testMoveAssignContextNull(void);
 void testMoveConstructContextNonNull(void);
 void testMoveConstructContextNull(void);
+namespace cl{
 MAKE_MOVE_TESTS(Context, make_context, clReleaseContext, contextPool)
-
+}
```

Comment:
This is a little ugly.  How about this as an alternative:

1. Separate out the "type" and the "typestr", whereas we just have a single "type" currently.
2. Use `Context` as the "typestr" in this case, and e.g. `CommandBuffer` as the "typestr" in the command buffer cases.
3. Use `cl::Context` as the "type" in this case, and e.g. `cl::khr::CommandBuffer` as the "type" in the command buffer cases.

I think this will require fewer diffs, and it won't require putting the tests themselves inside of the `cl` or `cl::khr` namespaces, which they really shouldn't be doing.

Reply:
@bashbaug In last commit (73c5491 https://github.com/KhronosGroup/OpenCL-CLHPP/pull/244/commits/73c5491d2497140d2c61352c49edccfd0e9dfa00) I have added corrections and removed not needed style corrections.